src/compiler/moduleNameResolver.ts

```diff
@@ -1677,10 +1704,11 @@ function nodeModuleNameResolverWorker(features: NodeResolutionFeatures, moduleNa
         failedLookupLocations,
         affectingLocations,
         diagnostics,
-        state.resultFromCache
+        state.resultFromCache,
+        legacyResult,
     );
 
-    function tryResolve(extensions: Extensions): SearchResult<{ resolved: Resolved, isExternalLibraryImport: boolean }> {
+    function tryResolve(extensions: Extensions, state: ModuleResolutionState): SearchResult<{ resolved: Resolved, isExternalLibraryImport: boolean }> {

```

Comment:
I missed why this parameter was needed. Is it an efficiency improvement to avoid relying on closure?